### PR TITLE
Avoid case sensitivity when handling baggage

### LINKED ISSUE #484: Case sensitivity affects OT propagator baggage handling
[Here](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/propagator/opentelemetry-propagator-ot-trace/src/opentelemetry/propagators/ot_trace/__init__.py#L98) is the problem. `startswith` is not case insensitive.

### PR DESCRIPTION
**Pull request by ocelotl**
Fixes #484

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #484

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A

# Does This PR Require a Core Repo Change?

- [ ] Yes. - Link to PR: 
- [X] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [ ] Followed the style guidelines of this project
- [ ] Changelogs have been updated
- [ ] Unit tests have been added
- [ ] Documentation has been updated

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   ([#472](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/472))
 - Set the `traced_request_attrs` of FalconInstrumentor by an argument correctly.
   ([#473](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/473))
+- `opentelemetry-propagator-ot-trace` Fix case sensitivity when handling baggage
+  ([#484](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/484))
```
Comment (lzchen) [inline CHANGELOG.md:16]:
```suggestion
  ([#485](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/485))
```

**Comment by lzchen** [comment]
Does this mean "oT-BaGgAgE-" is a valid prefix for a header? Can we not be explicit about it?

**Comment by ocelotl** [comment]
> Does this mean "oT-BaGgAgE-" is a valid prefix for a header? Can we not be explicit about it?

Hm, to say it is valid would be a bit too much of a general statement. With this PR I just mean to say that we will not care about casing when checking the prefixes of a baggage dictionary. This is done because some of the frameworks that handle the baggage headers have their own casing conventions and will transform `ot-baggage` into `Ot-Baggage`, for example.

What do you mean with "can we not be explicit about it"?, @lzchen

**Comment by ocelotl** [comment]
> Does this mean "oT-BaGgAgE-" is a valid prefix for a header? Can we not be explicit about it?

Hm, to say it is valid would be a bit too much of a general statement. With this PR I just mean to say that we will not care about casing when checking the prefixes of a baggage dictionary. This is done because some of the frameworks that handle the baggage headers have their own casing conventions and will transform `ot-baggage` into `Ot-Baggage`, for example.

What do you mean with "can we not be explicit about it"?, @lzchen

**Comment by lzchen** [comment]
> This is done because some of the frameworks that handle the baggage headers have their own casing conventions and will transform `ot-baggage` into `Ot-Baggage`,

So this use case makes sense, but with the current implementation `"oT-BaGgAgE-"` would be accepted as well correct? I'm not saying that this is wrong. If we accept that we consider that as a valid header then I don't have a problem with it.

**Comment by ocelotl** [comment]
> > This is done because some of the frameworks that handle the baggage headers have their own casing conventions and will transform `ot-baggage` into `Ot-Baggage`,
> 
> So this use case makes sense, but with the current implementation `"oT-BaGgAgE-"` would be accepted as well correct? I'm not saying that this is wrong. If we accept that we consider that as a valid header then I don't have a problem with it.

Hm, now that I think about this, this fix is not right, marking this as a draft for the time being.

**Comment by ocelotl** [comment]
After some more consideration, I think this is not the right approach, the code as it is right now is fine. Instead of making the handling of character cases in `extract` this is a job for the `getter.keys` method. Closing this PR.

**Pull request closed without merging**